Workiva/Color
# HY-5026 fix type problems when compiling with DDC

# Overview
The color repo doesn't compile with the Dart Development Compiler.

# Solution
Fix the dynamic-ish filter functions to convert to the correct types that are expected. They will have already been converted with the _convert function, but this makes the compiler happy.

# Testing
 - Use Dart >= 1.24.0 with DDC
 - `pub serve --web-compiler=dartdevc demo`
 - load localhost:8080 with regular Chrome, not Dartium. It should load the demo page.
